// angular2-wizard/__ivy_ngcc__/dist/index.js
"use strict";
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./src/wizard.component');
var ɵngcc2 = require('./src/wizard-step.component');
var ɵngcc3 = require('@angular/common');
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var wizard_component_1 = require("./src/wizard.component");
var wizard_step_component_1 = require("./src/wizard-step.component");
__export(require("./src/wizard.component"));
__export(require("./src/wizard-step.component"));
var FormWizardModule = FormWizardModule_1 = (function () {
    function FormWizardModule() {
    }
    FormWizardModule.forRoot = function () {
        return {
            ngModule: FormWizardModule_1
        };
    };
FormWizardModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FormWizardModule });
FormWizardModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FormWizardModule_Factory(t) { return new (t || FormWizardModule)(); }, imports: [[
            common_1.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FormWizardModule, { declarations: [ɵngcc1.WizardComponent, ɵngcc2.WizardStepComponent], imports: [ɵngcc3.CommonModule], exports: [ɵngcc1.WizardComponent, ɵngcc2.WizardStepComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FormWizardModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [
                    common_1.CommonModule
                ],
                declarations: [
                    wizard_component_1.WizardComponent,
                    wizard_step_component_1.WizardStepComponent
                ],
                exports: [
                    wizard_component_1.WizardComponent,
                    wizard_step_component_1.WizardStepComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return FormWizardModule;
}());
exports.FormWizardModule = FormWizardModule;
var FormWizardModule_1;
